snap/snapcraft.yaml

```diff
@@ -179,6 +179,23 @@ apps:
       - network
 
 parts:
+  microceph-orch:
+    plugin: python
+    # build-snaps:
+    #   - astral-uv
+    source: ./microceph-orch
+    after:
+      - microceph
+      - ceph
+    override-prime: |
+      set -eux
+      a=$(pwd)
+      craftctl default
+      mkdir -p share/ceph/mgr/microceph
+      # Force copy to overwrite.
+      cp -rf lib/python3.12/site-packages/microceph share/ceph/mgr/
+      cp -rf lib/python3.12/site-packages/orchestrator share/ceph/mgr/
```

Comment:
I'm a little wary of including / copying the orchestrator module here. I believe that this is included due to the need to enable microceph orchestrator as a valid orchestrator. Instead, I propose we apply a patch so that we don't have to refresh the orchestrator module routinely - otherwise we may miss out or remove upstream fixes to this python package.

Reply:
https://github.com/canonical/microceph/pull/584/files#diff-9f7543f993148cf2fef1b5d3905f48427441a692c5b21910e2b43bc7fc81e082R354

Reply:
for the moment, we have moved to consuming the modified orchestrator from a ppa.

Reply:
I think the PPA works (as you implied its for the moment) - but a patch file being applied is probably 100x simpler to maintain for the time being.

I have a patch with a proposal for this [here](https://github.com/wolsen/microceph/commit/847c1b4da3c180eb1064f7c379f2ad3f17354650) - though it's not a PR because I haven't had time to do more than verify the resulting snap file is correct.

Reply:
well I must say, using patches in snapcraft build itself is kinda cool. I have verified your commit. It worked! so i stole it :D